### DISCUSSION TITLE
Roll up of related accounts?

### DISCUSSION THREAD

**Discussion by BTWILS** [SFDO-Community/declarative-lookup-rollup-summaries#1143] [General]
Is it possible to roll up between two accounts related by a lookup? I have records of the account type of Partner that are associated with records with account type of Customer. I'd like to roll up a Value of Account field on the Customer record to a Total Value of Related Accounts field on the Partner record. 

Current setup that's erroring out:

**Lookup details**
Parent: Account
Child: Account
Relationship Field: ID
Relationship criteria fields: Total_Value_of_Opportunities__c (this is a rollup from opps related to the child account, and is working correctly)

**Rollup Details**
Field to Aggregate: Total_Value_of_Opportunities__c
Aggregate Operation: Sum
Aggregate Result Field: Total_Value_of_Related_Accounts__c

Error: **dlrs_AccountTrigger: execution of AfterInsert caused by: System.QueryException: Didn't understand relationship 'null' in field path. If you are attempting to use a custom relationship, be sure to append the '__r' after the custom relationship name. Please reference your WSDL or the describe call for the appropriate names. (dlrs)**

Any ideas?

**Comment by MelanieTolomeo** [comment]
Hi @BTWILS - try Relationship Field = ParentId

Ref: https://developer.salesforce.com/docs/atlas.en-us.object_reference.meta/object_reference/sforce_api_objects_account.htm

**Reply by BTWILS** [reply]
Thanks @MelanieTolomeo Is there a way to do this without using account hierarchies? We have actual organizational hierarchies, ie FedEx and its subsidiaries. The model I want to implement is a reseller account and its clients. Also, if one of these client accounts is part of an account hierarchy, how to we honor both relationships?

**Reply by MelanieTolomeo** [reply]
Hi @BTWILS - what is the field that relates the two accounts called? do you have its API name?

And do you want to filter the child records in some way (e.g. for only customer accounts)? If so, you can do that with a SOQL query in the Relationship Criteria field.
